packages/bee-q/src/components/option/bq-option.tsx

```diff
@@ -0,0 +1,170 @@
+import { h, Component, Prop, Element, State, Event, EventEmitter, Listen } from '@stencil/core';
+
+import { hasSlotContent } from '../../shared/utils';
+
+/**
+ * @part base - The component's internal wrapper.
+ * @part label - The `span` element in which the label text is displayed.
+ * @part prefix - The `span` element in which the prefix is displayed (generally `bq-icon`).
+ * @part suffix - The `span` element in which the suffix is displayed (generally `bq-icon`).
+ */
+@Component({
+  tag: 'bq-option',
+  styleUrl: './scss/bq-option.scss',
+  shadow: true,
+})
+export class BqOption {
+  // Own Properties
+  // ====================
+
+  private prefixElem: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqOptionElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() hasPrefix: boolean = false;
+
+  // Public Property API
+  // ========================
+
+  /** If true, the option is disabled. */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+
+  /** A string representing the value of the option. Can be used to identify the item */
+  @Prop({ reflect: true }) value?: string;
+
+  /** If true, the option is selected and active. */
+  @Prop({ reflect: true }) selected: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called when item loses focus */
+  @Event() bqOptionBlur: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called when item is focused */
+  @Event() bqOptionFocus: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called when item is clicked */
+  @Event() bqOptionClick: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called on enter key press */
+  @Event() bqOptionEnter: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called on enter key up */
+  @Event() bqOnEnterKeyUp: EventEmitter<HTMLBqOptionElement>;
```

Comment:
We should be consistent with our naming conventions for custom events:

* bqBlur
* bqFocus
* bqClick
* bqEnter
* bqKeyUp